pkg/minikube/machine/info.go

```diff
@@ -97,3 +99,45 @@ func logRemoteOsRelease(r command.Runner) {
 
 	glog.Infof("Remote host: %s", osReleaseInfo.PrettyName)
 }
+
+var cachedSystemMemoryLimit *mem.VirtualMemoryStat
+var cachedSystemMemoryErr *error
+
+//  cachedSysMemLimit will return a chaced limit for the system's virtual memory.
+func cachedSysMemLimit() (*mem.VirtualMemoryStat, error) {
+	if cachedSystemMemoryLimit == nil {
+		v, err := mem.VirtualMemory()
+		cachedSystemMemoryLimit = v
+		cachedSystemMemoryErr = &err
+	}
+	return cachedSystemMemoryLimit, *cachedSystemMemoryErr
```

Comment:
Do we really want to cache the error? What if the error isn't persistent?

Reply:
yeah, each time it is called a new error is cached. this is good so when we return a cached item, we can handle it with the same cached error